src/Components/CreateImageWizardV2/utilities/requestMapper.tsx

```diff
@@ -92,14 +93,35 @@ export const mapRequestFromState = (
   };
 };
 
+const parseSizeUnit = (bytesize: string) => {
+  let size;
+  let unit;
+
+  if (parseInt(bytesize) % UNIT_GIB === 0) {
+    size = parseInt(bytesize) / UNIT_GIB;
+    unit = 'GiB' as Units;
```

Comment:
Can you change to `unit: Units = 'GiB'`? Using `as` is like a wrecking ball, it will allow you to do total nonsense. If you use what I proposed type checking will actually occur.